near-sdk/src/store/lazy/mod.rs

```diff
@@ -144,6 +144,137 @@ where
     }
 }
 
+/// An persistent lazily loaded option, that stores a value in the storage.
+///
+/// This will only write to the underlying store if the value has changed, and will only read the
+/// existing value from storage once.
+///
+/// # Examples
+/// ```
+/// use near_sdk::store::LazyOption;
+///
+///# near_sdk::test_utils::test_env::setup();
+/// let mut a = LazyOption::new(b"a", None);
+/// assert_eq!(*a, None);
+///
+/// *a = Some("new value".to_owned());
+/// assert_eq!(a.get(), &Some("new value".to_owned()));
```

Comment:
maybe worth updating these docs to use as an option? for example for the first line:
```suggestion
/// assert!(a.is_none());
///
/// *a = Some("new value".to_owned());
/// assert_eq!(a.get(), &Some("new value".to_owned()));
```

And can maybe do a few Option things like `take` or mapping?